pkgs/by-name/gi/git-branchless/package.nix

```diff
@@ -21,6 +21,21 @@ rustPlatform.buildRustPackage rec {
     hash = "sha256-8uv+sZRr06K42hmxgjrKk6FDEngUhN/9epixRYKwE3U=";
   };
 
+  # Patch the vendored esl01-indexedlog crate.
+  # This is necessary to fix the build for rust 1.89. See:
+  # - https://github.com/NixOS/nixpkgs/issues/437051
+  # - https://github.com/arxanas/git-branchless/issues/1585
+  # - https://github.com/facebook/sapling/issues/1119
+  # The patch is derived from:
+  # - https://github.com/facebook/sapling/commit/9e27acb84605079bf4e305afb637a4d6801831ac
+  prePatch = ''
+    pushd ../git-branchless-*-vendor/esl01-indexedlog-*/
+  '';
+  patches = [
+    ./fix-esl01-indexedlog-for-rust-1_89.patch
+  ];
+  postPatch = "popd";
```

Comment:
I would rather do the patching entirely in postPatch to not break patches for "normal" patches

Reply:
Good idea! Fixed and force-pushed!